File: docs/src/testers/functions/invocations/handling_results.ts
// @sniptest filename=handling_results.ts
import { NotteClient } from 'notte-sdk';

const client = new NotteClient();
const fn = client.NotteFunction({
  function_id: process.env.NOTTE_FUNCTION_ID!,
});

// Check the final status and structured return value.
const result = await fn.run({ url: 'https://example.com' });
console.log(JSON.stringify({ status: result.status, result: result.result }));

File: docs/src/testers/functions/invocations/handling_results.py
# @sniptest filename=handling_results.py
import json
import os

from notte_sdk import NotteClient

client = NotteClient()
function = client.Function(os.environ["NOTTE_FUNCTION_ID"])

# Check the final status and structured return value.
result = function.run(url="https://example.com")
print(json.dumps({"status": result.status, "result": result.result}))
